.github/workflows/ci.yaml

```diff
@@ -36,11 +36,17 @@ jobs:
           - "3.8"
           - "3.9"
           - "3.10"
+          - "3.11"
+          - "3.12"
         include:
           - os: "ubuntu-20.04"
             python: "3.6"
           - os: "ubuntu-20.04"
             python: "pypy-3.6"
+          - os: "ubuntu-latest"
+            python: "pypy-3.10"
+          - os: "macos-latest"
+            python: 3.x
```

Comment:
Let's use explicit `ubuntu-22.04` and `macos-12` here please, the more pinned, the better for CI stability. Can we pin `3.x` any more? Is this intended to be 3.13 pre-release?

Reply:
My Mac is running 14.1.1.  Why do we want to pin to an OS that is two major versions behind the current macOS?
* https://www.google.com/search?client=firefox-b-d&q=current+macos

Python `3.x` pins to the latest ___released___ version of CPython which will remain `3.12` until Oct. 2024.

I like it when CI highlights when OS upgrades break the tests rather than waiting for users to complain.